Invalid OAuth scope or ID token audience provided

### Question

I am using socket factory with "cloudSqlGoogleCredentialsPath" with Tableau but it is giving below error, where as if I use same json with cloudsql auth proxy then I am able to connect. please let me know issue

CONNECTION STRING -

jdbc:postgresql:///DATABASE_NAME?cloudSqlInstance=PROJECT_ID:REGION:INSTANCE_NAME&socketFactory=com.google.cloud.sql.postgres.SocketFactory&password=NotNull&enableIamAuth=true&sslmode=disable&cloudSqlGoogleCredentialsPath=WINDOWS_PATH_FOR_SA_JSON 

ERROR - 

Caused by: com.google.auth.oauth2.GoogleAuthException: Error getting access token for service account: 400 Bad Request
POST https://oauth2.googleapis.com/token
{"error":"invalid_scope","error_description":"Invalid OAuth scope or ID token audience provided."}, iss: tableau-gcp-dbg-sa@db-dev-x1hq-tableau-gcp-dbg.iam.gserviceaccount.com
	at com.google.auth.oauth2.GoogleAuthException.createWithTokenEndpointResponseException(GoogleAuthException.java:129) ~[?:?]
	at com.google.auth.oauth2.ServiceAccountCredentials.refreshAccessToken(ServiceAccountCredentials.java:538) ~[?:?]
	at com.google.auth.oauth2.OAuth2Credentials$1.call(OAuth2Credentials.java:270) ~[?:?]
	at com.google.auth.oauth2.OAuth2Credentials$1.call(OAuth2Credentials.java:267) ~[?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at com.google.auth.oauth2.OAuth2Credentials$RefreshTask.run(OAuth2Credentials.java:635) ~[?:?]
	at com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30) ~[jdbcserver.jar:20213.0.10]
	at com.google.auth.oauth2.OAuth2Credentials$AsyncRefreshResult.executeIfNew(OAuth2Credentials.java:582) ~[?:?]
	at com.google.auth.oauth2.OAuth2Credentials.asyncFetch(OAuth2Credentials.java:233) ~[?:?]
	at com.google.auth.oauth2.OAuth2Credentials.refreshIfExpired(OAuth2Credentials.java:204) ~[?:?]
	at com.google.cloud.sql.core.DefaultAccessTokenSupplier.lambda$get$0(DefaultAccessTokenSupplier.java:87) ~[?:?]
	at com.google.cloud.sql.core.RetryingCallable.call(RetryingCallable.java:67) ~[?:?]
	at com.google.cloud.sql.core.DefaultAccessTokenSupplier.get(DefaultAccessTokenSupplier.java:140) ~[?:?]

### Code

```Java
No code change made, just build and using postgres socket factory
```


### Additional Details

_No response_

--- a/core/src/main/java/com/google/cloud/sql/core/FileCredentialFactory.java
+++ b/core/src/main/java/com/google/cloud/sql/core/FileCredentialFactory.java
@@ -17,11 +17,13 @@
 package com.google.cloud.sql.core;
 
 import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.services.sqladmin.SQLAdminScopes;
 import com.google.auth.http.HttpCredentialsAdapter;
 import com.google.auth.oauth2.GoogleCredentials;
 import com.google.cloud.sql.CredentialFactory;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.util.Arrays;
 
 class FileCredentialFactory implements CredentialFactory {
   private final String path;
@@ -37,10 +39,19 @@ class FileCredentialFactory implements CredentialFactory {
 
   @Override
   public GoogleCredentials getCredentials() {
+    GoogleCredentials credentials;
     try {
-      return GoogleCredentials.fromStream(new FileInputStream(path));
+      credentials = GoogleCredentials.fromStream(new FileInputStream(path));
     } catch (IOException e) {
       throw new IllegalStateException("Unable to load GoogleCredentials from file " + path, e);
     }
+
+    if (credentials.createScopedRequired()) {
+      credentials =
+          credentials.createScoped(
+              Arrays.asList(SQLAdminScopes.SQLSERVICE_ADMIN, SQLAdminScopes.CLOUD_PLATFORM));
+    }
+
+    return credentials;
   }
 }

